### PR TITLE
Adding support for ORCHARD_APP_DATA environment variable

### PR DESCRIPTION
**Pull request by sebastienros**
Setting `ORCHARD_APP_DATA` environment variable to `Foo` will store the tenants (and lucene indices, and media ,...) in the folder `[ContentRootPath]\Foo`.

Setting it to a UNC will work too, like `C:\temp\mytenants`.

We can now use a shared drive to store tenants, and use it in Docker. For Azure App Services Containers use these two variables:
`ORCHARD_APP_DATA=/home`
`WEBSITES_ENABLE_APP_SERVICE_STORAGE=true`

Then restarting the container, updating it, or activating several of them in the same app will just work.

### CODE REVIEW

**Comment by sebastienros** [comment]
@agriffard you might want to do some test to ensure I didn't break anything. I tried to setup new sites, and also played with Assets and Lucene. The environment variable can be set in the project properties is you start from VS.

**Comment by sebastienros** [comment]
@agriffard that's great, I tried it too, I think it's awesome to be able to do that.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs [hunk 1]
```diff
@@ -78,16 +78,16 @@ public override void ConfigureServices(IServiceCollection services)
             services.TryAddEnumerable(new[]
             {
                 // Orchard-specific initializers:
-                ServiceDescriptor.Singleton<IConfigureOptions<AuthenticationOptions>, OpenIdConfiguration>(),
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs]: @sebastienros hum, why changing the lifetime of the initializers?

**Comment by sebastienros** [comment]
I thought it was a typo as most setup services are run only once. Why
should it be a Singleton?

**Comment by PinpointTownes** [comment]
Well, it shouldn't matter much as the default `OptionsMonitor`/`OptionsManager` cache the options instances (so the initializers shouldn't be called twice).

I had opted for `Singleton` because that's what all the ASP.NET Core handlers do.

**Pull request merged**